### PR TITLE
GH-153: Publish to Maven Central

### PR DESCRIPTION
**Pull request by jmewes**

### CODE REVIEW

File: restdocs-api-spec-gradle-plugin/src/main/kotlin/com/epages/restdocs/apispec/gradle/OpenApiExtension.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ abstract class OpenApiBaseExtension(project: Project) : ApiSpecExtension(project
     }
 
     private fun scopeDescriptionSource(scopeDescriptionsPropertiesFile: File): Map<String, String> {
-        return scopeDescriptionsPropertiesFile.let { objectMapper.readValue<Map<String, String>>(it) } ?: emptyMap()
+        return scopeDescriptionsPropertiesFile.let { objectMapper.readValue(it) }
```
Comment (jmewes) [inline restdocs-api-spec-gradle-plugin/src/main/kotlin/com/epages/restdocs/apispec/gradle/OpenApiExtension.kt:40]: Fix of compilation warning. Done with auto refactorings from IntelliJ.

File: restdocs-api-spec-mockmvc/src/main/kotlin/com/epages/restdocs/apispec/MockMvcRestDocumentationWrapper.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ object MockMvcRestDocumentationWrapper : RestDocumentationWrapper() {
             enhanceSnippetsWithResourceSnippet(
                 resourceDetails = resourceDetails,
                 snippetFilter = snippetFilter,
-                snippets = *snippets
+                snippets = snippets
```
Comment (jmewes) [inline restdocs-api-spec-mockmvc/src/main/kotlin/com/epages/restdocs/apispec/MockMvcRestDocumentationWrapper.kt:30]: This an similar changes fixes compilation warnings about the unnecessary usage of the spread operator.

File: samples/restdocs-api-spec-sample-web-test-client/build.gradle [hunk 3]
```diff
@@ -33,7 +33,7 @@ ext {
 dependencies {
 	implementation('org.springframework.boot:spring-boot-starter-webflux')
 
-	testImplementation('org.junit.jupiter:junit-jupiter-engine:5.4.2')
+	testImplementation('org.junit.jupiter:junit-jupiter-engine')
```
Comment (jmewes) [inline samples/restdocs-api-spec-sample-web-test-client/build.gradle:36]: Without this change the build will fail because JUnit cannot find any test.

**Comment by jmewes** [comment]
I have manually created a staging repository from a WIP commit which is almost the same like the one in the PR. Here all the requirements for closing the staging repository where matched. This means, that there should be nothing in the way to publish it.

Nexus_Repository_Manager

**Comment by jmewes** [comment]
In a WIP commit which forces a triggering of the publish script from a PR, I could successfully create a SNAPSHOT release. This means that if there is a tag on the current commit, it should successfully create a staging repository.

See https://github.com/ePages-de/restdocs-api-spec/pull/159#issuecomment-799464412

**Comment by jmewes** [comment]
I fixed 90% of the compilation warnings which came along with the required Gradle and Kotlin upgrades but I left some of some as an excersise for the reader:

```
> Task :restdocs-api-spec-gradle-plugin:test
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/Users/jmewes/.gradle/wrapper/dists/gradle-6.8.3-bin/7ykxq50lst7lb7wx1nijpicxn/gradle-6.8.3/lib/groovy-all-1.3-2.5.12.jar) to method java.lang.Object.clone()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

> Task :restdocs-api-spec:compileKotlin
w: /Users/jmewes/src/epages-beyond/restdocs-api-spec/restdocs-api-spec/src/main/kotlin/com/epages/restdocs/apispec/ResourceSnippet.kt: (66, 24): Type mismatch: inferred type is String? but String was expected

> Task :restdocs-api-spec-openapi-generator:compileKotlin
w: /Users/jmewes/src/epages-beyond/restdocs-api-spec/restdocs-api-spec-openapi-generator/src/main/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20Generator.kt: (318, 17): 'when' expression on enum is recommended to be exhaustive, add 'JWT_BEARER' branch or 'else' branch instead

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 3m 12s
84 actionable tasks: 81 executed, 3 up-to-date


```

File: gradlew [hunk 4]
```diff
@@ -125,10 +126,11 @@ if $darwin; then
     GRADLE_OPTS="$GRADLE_OPTS \"-Xdock:name=$APP_NAME\" \"-Xdock:icon=$APP_HOME/media/gradle.icns\""
 fi
 
-# For Cygwin, switch paths to Windows format before running java
-if $cygwin ; then
+# For Cygwin or MSYS, switch paths to Windows format before running java
+if [ "$cygwin" = "true" -o "$msys" = "true" ] ; then
```
Comment (andreykletushkin) [inline gradlew:130]: Did you upgrade gradlew to be compatible with maven?

Comment (jmewes) [reply gradlew:130]: No, I upgraded the Gradle versions to be able to use the "publish-plugin" and current version of the "java" plugin.

- https://github.com/ePages-de/restdocs-api-spec/blob/18d8af911eec7e24a07aeef50e3079735ae9909c/build.gradle.kts#L10
- https://github.com/ePages-de/restdocs-api-spec/blob/18d8af911eec7e24a07aeef50e3079735ae9909c/restdocs-api-spec/build.gradle.kts#L74-L77

Comment (jmewes) [reply gradlew:130]: And using a recent version is always good ;)

**Review by andreykletushkin — APPROVED** [review]
LGTM

**Pull request merged**